src/app/(with-header-sidebar)/mypage/_components/PasswordForm.tsx

```diff
@@ -16,18 +20,43 @@ export default function PasswordForm() {
   const {
     register,
     handleSubmit,
-    formState: { errors },
+    formState: { errors, isValid },
     watch,
     trigger,
     setError,
     reset,
   } = useForm<PasswordFormValues>({ mode: 'onChange' });
+  const { openModal } = useModalStore();
 
   const customIsValid = Object.keys(errors).length === 0;
   const watchedPassword = watch('newPassword');
 
-  const onSubmit = async (data: PasswordFormValues) =>
-    updatePassword(data, reset, setError);
+  const onSubmit = async (data: PasswordFormValues) => {
+    try {
+      await axios.put('/auth/password', {
+        password: data.currentPassword,
+        newPassword: data.newPassword,
+      });
+      reset();
+    } catch (error) {
+      if (error instanceof Error) {
+        if (error.message === ERROR_MESSAGES.CURRENT_PASSWORD_INCORRECT) {
+          openModal(<AlertModal>{error.message}</AlertModal>);
+          setError('currentPassword', {
+            type: 'manual',
+            message: error.message,
+          });
+        }
+        if (error.message === ERROR_MESSAGES.SAME_AS_OLD_PASSWORD) {
+          openModal(<AlertModal>{error.message}</AlertModal>);
+          setError('newPassword', {
+            type: 'manual',
+            message: error.message,
+          });
+        }
```

Comment:
몬가..if문안에 if문은 지양하는 느낌이여서
switch 문을 쓰면 더 보기 편할것같아요!
아니면 요런느낌도 좋을것같아요~!
```suggestion
const errorHandlingMap = {
  [ERROR_MESSAGES.CURRENT_PASSWORD_INCORRECT]: 'currentPassword',
  [ERROR_MESSAGES.SAME_AS_OLD_PASSWORD]: 'newPassword',
};

const field = errorHandlingMap[error.message];
if (field) {
  openModal(<AlertModal>{error.message}</AlertModal>);
  setError(field, {
    type: 'manual',
    message: error.message,
  });
}

```